### DISCUSSION TITLE
Getting data from database

### DISCUSSION THREAD

**Discussion by edvard-bjarnason** [reflex-dev/reflex#3130] [General]
I have a question, how do you easily get data from the database so that it can be processed?

with rx.session() as session:
            self.lines = session.exec(select(line)).all()

returns Var that is is difficult to iterate over and process further

**Comment by picklelo** [comment]
Is this within your event handler? In the backend everything should be a normal Python object that you can process normally. It should only be a `Var` when referenced in your frontend code.

**Reply by edvard-bjarnason** [reply]
Yes exactly the code was in an event_handler (a function of a class extending rx.State) then I want to pass the processed data to a class extending rx.Component. 

The solution that works for me was to prefix the function in the rx.State derived class with _ not sure this is the most elegant way?

Where do you recommend to put code like this, code that gets some data from database and processes it (maybe filter and format the data) and then use the processed data as state in a component?

Thanks.

**Reply by picklelo** [reply]
Anything that doesn't involve state you could just make as functions outside of the state as well, and just call it from your event handler. I'm curious though what your code that threw an error was - it should work without needing the _ in front.
